### PR TITLE
Extract equal functionality for config items to traits

### PR DESCRIPTION
**Pull request by EPashkin**
Trait case for #211

### CODE REVIEW

File: src/analysis/functions.rs [hunk 1]
```diff
@@ -9,6 +9,7 @@ use analysis::return_value;
 use analysis::rust_type::*;
 use analysis::safety_assertion_mode::SafetyAssertionMode;
 use config;
+use config::identables::Identables;
```
Comment (gkoz) [inline src/analysis/functions.rs]: It might make sense to reexport `Identables` in `mod traits`;

File: src/config/identables.rs [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    fn matched(&self, name: &str) -> Vec<&Self::Item>;
+}
+
+impl<T: Parse + AsRef<Ident>> Identables for Vec<T> {
+    type Item = T;
+    
+    fn parse(toml: Option<&Value>, object_name: &str) -> Vec<Self::Item> {
+        let mut v = Vec::new();
+        if let Some(fns) = toml.and_then(|val| val.as_slice()) {
+            for f in fns {
```
Comment (gkoz) [inline src/config/identables.rs]: More generic names for `f` and `fns` would be better.

**Comment by gkoz** [comment]
I like this better than macros.

**Comment by EPashkin** [comment]
Fixed

**Comment by gkoz** [comment]
> It might make sense to reexport `Identables` in mod `traits`

Not sure if you disagree or misunderstood. What I meant is we already import `traits::*` so the `Identables` trait could get imported for free.

**Comment by EPashkin** [comment]
I missed about `mod traits`.

**Comment by EPashkin** [comment]
Reexported

**Comment by EPashkin** [comment]
I also prefer this variant

**Comment by gkoz** [comment]
Thanks!

**Pull request merged**